SteamKit2/SteamKit2/Types/DepotManifest.cs

```diff
@@ -83,36 +86,50 @@ public class FileData
             /// <summary>
             /// Gets the name of the file.
             /// </summary>
-            public string FileName { get; internal set; }
+            public string FileName { get; set; }
             /// <summary>
             /// Gets SHA-1 hash of this file's name.
             /// </summary>
-            public byte[] FileNameHash { get; internal set; }
+            public byte[] FileNameHash { get; set; }
             /// <summary>
             /// Gets the chunks that this file is composed of.
             /// </summary>
-            public List<ChunkData> Chunks { get; private set; }
+            public List<ChunkData> Chunks { get; set; }
 
             /// <summary>
             /// Gets the file flags
             /// </summary>
-            public EDepotFileFlag Flags { get; private set; }
+            public EDepotFileFlag Flags { get; set; }
 
             /// <summary>
             /// Gets the total size of this file.
             /// </summary>
-            public ulong TotalSize { get; private set; }
+            public ulong TotalSize { get; set; }
             /// <summary>
             /// Gets SHA-1 hash of this file.
             /// </summary>
-            public byte[] FileHash { get; private set; }
+            public byte[] FileHash { get; set; }
             /// <summary>
             /// Gets symlink target of this file.
             /// </summary>
-            public string LinkTarget { get; private set; }
+            public string LinkTarget { get; set; }
 
+            /// <summary>
+            /// Initializes a new instance of the <see cref="FileData"/> class.
+            /// </summary>
+            public FileData()
+            {
+                FileName = "";
+                FileNameHash = [];
+                Chunks = new List<ChunkData>();
+                FileHash = [];
+                LinkTarget = "";

```

Comment:
Isn't it null by default in the protos?

Reply:
Technically, all fields are null by default in the protos. This particular field is optional if that's what you mean, it's usually not set in official manifests. `DepotManifest::Serialize` checks IsNullOrWhiteSpace so it's safe to use empty string here.

Reply:
I would just avoid setting it